### DISCUSSION TITLE
Turning Helix into an IDE with the help of WezTerm and CLI tools

### DISCUSSION THREAD

**Discussion by quantonganh** [helix-editor/helix#8000] [Show and tell]
https://quantonganh.com/2023/08/19/turn-helix-into-ide

Please let me know if you have anything to add. Thank you.

**Comment by zetashift** [comment]
I've been using [Zellij](https://zellij.dev/news/floating-panes-tmux-mode/) for this instead of wezterm.

But I wholehearthedly recommend people try out the things recommended in the article. It's a great replacement for plugins for somethings(like `lazygit`!).

**Reply by Mimry** [reply]
Sounds interesting, would mind sharing how you achieved that with Zellij?

**Reply by zetashift** [reply]
TLDR: use `:sh zellij run -- my-awesome-cli-tool` to get functionality you might want in a plugin, that's replaced by `my-awesome-cli-tool`.

---

Here's a snippet for using lazygit:
```toml
[keys.normal.space.g]
g = ":sh zellij action new-tab -l lg"
s = ":sh zellij run -fc -- lazygit"
```

Or for searching through my notes:
```
[keys.normal.space.n]
f = ":sh zellij action new-tab -l notes"
```

I have cobbled up some `nushell` scripts that use `fzf` to fuzzy find in my notes, or capture some note in a new zellij floating window. And these scripts are then binded to a key as well.

That's about it actually. Helix does 80% of what I want ootb anyway.

**Reply by quantonganh** [reply]
I actually tried something like this:

```sh
#!/bin/sh

zellij action new-pane -f
for ((i=1; i<=11; i++))
do
  zellij action resize +
done

zellij action write-chars "lazygit"
zellij action write 13
```

since I enjoy the [floating panes](https://zellij.dev/news/floating-panes-tmux-mode/) feature. However, I didn't want to add another layer to my terminal, so I reverted to using WezTerm.

**Reply by zetashift** [reply]
Yea if wezterm had floating panes I'd just switch to that

**Comment by georgealexanderday** [comment]
Currently getting an Error code: PR_END_OF_FILE_ERROR when trying to access your website but would love to read it.

**Reply by quantonganh** [reply]
@georgealexanderday: It's strange. Are you using Firefox? Have you tried with other browsers? Alternatively, you can read it [on GitHub](https://github.com/quantonganh/blog/blob/master/2023/08/19/turn-helix-into-ide.md).

**Reply by georgealexanderday** [reply]
strange yeah it was on Firefox but seems to be working today, sorry if any panic caused!

**Comment by quantonganh** [comment]
I've just relocated this to [a separate repository](https://github.com/quantonganh/helix-wezterm), making it easier for everyone to install and/or contribute.
